internal/http/monitor_runner_health_test.go

```diff
@@ -0,0 +1,149 @@
+package http
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"os/exec"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestSendRunnerHealthCheckRequest(t *testing.T) {
+	successSrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer successSrv.Close()
+
+	err := sendRunnerHealthCheckRequest(successSrv.URL)
+	if err != nil {
+		t.Errorf("Expected no error, got %v", err)
+	}
+
+	errorSrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(http.StatusInternalServerError)
+	}))
+	defer errorSrv.Close()
+
+	err = sendRunnerHealthCheckRequest(errorSrv.URL)
+	if err == nil {
+		t.Error("Expected error for non-200 status code, got nil")
+	}
+}
+
+func setHealthCheckValues(t *testing.T) func() {
+	t.Helper()
+
+	originalHealthCheckInterval := healthCheckInterval
+	originalInitialDelay := initialDelay
+	originalMaxFailures := healthCheckMaxFailures
+
+	healthCheckInterval = 50 * time.Millisecond
+	initialDelay = 0
+	healthCheckMaxFailures = 2
+
+	return func() {
+		healthCheckInterval = originalHealthCheckInterval
+		initialDelay = originalInitialDelay
+		healthCheckMaxFailures = originalMaxFailures
+	}
+}
+
+func TestMonitorRunnerHealth(t *testing.T) {
+	restoreFn := setHealthCheckValues(t)
+	defer restoreFn()
+
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer srv.Close()
+
+	cmd := exec.Command("sleep", "1")
+	if err := cmd.Start(); err != nil {
+		t.Fatal(err)
+	}
+
+	defer func() {
+		if cmd.Process != nil && cmd.Process.Kill() != nil {
+			t.Log("Failed to kill process during cleanup")
+		}
+	}()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+	var wg sync.WaitGroup
+
+	MonitorRunnerHealth(ctx, cmd, srv.URL, &wg)
```

Comment:
Maybe we should improve the interface for this function to either return a value that tells either a) monitoring failed b) it was cancelled OR use a channel where it can post if the monitoring fails. That way we can extract the killing of the process outside the actual monitoring and it would be much easier to test

Reply:
You're right! After sleeping on it, this test is overly complicated because the function itself is not easy to test. I'll look into this.